pkg/image/cache.go

```diff
@@ -77,6 +77,12 @@ func newCacheProxy() *cacheProxy {
 	}
 }
 
+func (c *cacheProxy) ClearCache() {
```

Comment:
```suggestion
func (c *cacheProxy) storeGlobalPullSecret(_ []byte) {
```

this way and by moving storeGlobalPullSecret from the RegistryInspector to ICache interface, the logic of storeGlobalPullSecret is left to the specific cache implementation, with the Purge of the cache being an internal-to-this-package optimization choice as we were discussing, hidden from the rest of the code that can eventually make use of other implementations of RegistryInspector/Cache pairs (e.g. for the new k8s SA-based auth feature I sent yesterday).

Reply:
yup i understand the change - but don't really like that we are using `storeGlobalPullSecret` to clear the cache - it's kinda confusing as it has nothing to do with caching - it satisfies the pattern, but maybe worth revisting to see if this can be structured better to accomodate more functions without violating the pattern

Reply:
> by moving storeGlobalPullSecret from the RegistryInspector to ICache interface
this will cause a problem in the testing framework because ICache will have an unexported method and even if we change the fake cache to be CacheProxy type rather than ICache, the assignment [here](https://github.com/openshift/multiarch-tuning-operator/blob/main/controllers/podplacement/pod_model_test.go#L282) fails since that is also used to assign it to an ICache object later.